### DISCUSSION TITLE
Using the trace-filter tool

### DISCUSSION THREAD

**Discussion by poldrack** [quarto-dev/quarto-cli#7012] [Q&A]
### Description

I am trying to debug a Lua filter using the `quarto-filter-trace.json` file generated via `QUARTO_TRACE_FILTERS=true`.  The filter trace viewer does not seem to act as suggested by the documentation (https://quarto.org/docs/troubleshooting/#debugging-lua-filters).  In the latest version of quarto-cli (commit 00512ab2) I see `tools/trace-viewer` but it's not clear how to use it; passing the input file as suggested in the outdated documentation doesn't work.  Can you provide some documentation on how to run this viewer?

**Comment by cscheid** [comment]
Sorry - that documentation is in fact out of date!

Try this:

```
$ QUARTO_TRACE_FILTERS=output.json quarto render test.qmd
$ cd ...quarto-cli/tools/trace-viewer
$ quarto preview trace-viewer.qmd
```

Then, drag the .json file into the "Trace 1" or "Trace 2" text fields.

**Reply by cscheid** [reply]
I just updated the documentation as well; thanks for letting us know!

**Comment by poldrack** [comment]
Many thanks! It might also be good to add something to the rendered trace viewer page like "drag output.json file here to get started" - I had actually seen this page earlier but didn't realize I needed to drag and drop the file.

**Reply by cscheid** [reply]
That .qmd needs a lot of work (and I should know it, I wrote it). I developed it for internal use, but troubleshooting.qmd is publicly visible, so there's no excuse. We'll improve it.

**Comment by poldrack** [comment]
great, thanks! closing but feel free to reopen.
